tests/gpflow/utilities/test_deepcopy.py

```diff
@@ -87,6 +87,21 @@ def test_freeze() -> None:
     assert module_frozen.variables == ()
     assert isinstance(module.module.module.var, tf.Variable)
     assert isinstance(module_frozen.module.module.var, tf.Tensor)
+    assert module.module.module.var.dtype is tf.float64
+    assert module_frozen.module.module.var.dtype is tf.float64
+    assert module.module.module.var == module_frozen.module.module.var
+
+
+def test_freeze_as_float32() -> None:
+    module = NestedModule(NestedModule(A()))
```

Comment:
Q: Why NestedModule twice?

Reply:
To check that freeze_as_float32 recursively parses the module to find all the tensors (rather than just looking at the top level).